polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -250,9 +236,14 @@ struct Pointer2MemrefOpLowering
     // MemRefDescriptor sourceMemRef(operands.front());
     auto convertedType = getTypeConverter()->convertType(op.getType());
     assert(convertedType && "unexpected failure in memref type conversion");
+    auto ptr = adaptor.source();
+    auto ptrTy = ptr.getType().cast<LLVM::LLVMPointerType>();
+    if (ptrTy.getAddressSpace() != 0)
```

Comment:
I would check that the source and destination pointer types have to be in the same address space rather than assuming the dest. ptr type is in address space zero.